### PR TITLE
Add CDA + Common Platform Systems

### PR DESCRIPTION
**Pull request by richrace**
## What does this pull request do?

Added Court Data Adapter and Common Platform systems

## What is the intent behind these changes?

Document how the CDA interacts with CP and VCD.

### Landscape

structurizr-55246-system-overview

### System

structurizr-55246-cda-context

### Container

structurizr-55246-cda-container

### CODE REVIEW

File: src/main/kotlin/model/VCD.kt [hunk 1]
```diff
@@ -52,6 +52,9 @@ class VCD private constructor() {
       // user relationships
       LegalAidAgencyUsers.crimeApplicationCaseWorker.uses(web, "Searches and links/unlinks defendants to MAAT")
       LegalAidAgencyUsers.billingCaseWorker.uses(web, "Searches and inspects defendants' case hearing history")
+
+      // declare relationships to other systems and other system containers
+      web.uses(CDA.api, "Provides interface to HMCTS Common Platform", "REST")
```
Comment (heidar) [inline src/main/kotlin/model/VCD.kt:57]: Is it possible to summarise what data are being passed here?

**Review by heidar — APPROVED** [review]
Just left that one question, looks great. :)

File: src/main/kotlin/model/VCD.kt [hunk 2]
```diff
@@ -52,6 +52,9 @@ class VCD private constructor() {
       // user relationships
       LegalAidAgencyUsers.crimeApplicationCaseWorker.uses(web, "Searches and links/unlinks defendants to MAAT")
       LegalAidAgencyUsers.billingCaseWorker.uses(web, "Searches and inspects defendants' case hearing history")
+
+      // declare relationships to other systems and other system containers
```
Comment (sldblog) [inline src/main/kotlin/model/VCD.kt:56]: 🙋‍♂️ _suggestion_

It looks like this comment is a recurring pattern -- what do you think about adding this back to the original interface as a separate method?

Maybe something like the below?
```diff
 interface LAASoftwareSystem {
   fun defineModelEntities(model: Model)
-  fun defineRelationships()
+  fun defineUserRelationships()
+  fun defineOtherSystemRelationships()
   fun defineViews(views: ViewSet)
 }
```
etc?

Comment (richrace) [reply src/main/kotlin/model/VCD.kt:56]: I agree with this. It would definitely help readability.

File: src/main/kotlin/model/VCD.kt (hunk 1 above)
Comment (richrace) [reply src/main/kotlin/model/VCD.kt]: Updated 👍

**Pull request merged**